### ISSUE TITLE
Reflow: Instances of two-dimensional scroll

### ISSUE THREAD

**Issue by GCHQ-Developer-299** [mi6/ic-design-system#1706]
### Summary of the bug

In order to fully meet [WCAG Criteria 1.4.10 Reflow](https://www.w3.org/WAI/WCAG22/Understanding/reflow.html) we have to ensure none of our pages require horizontal scrolling at the designated viewport width (320px at 100% zoom, 1280px at 400% zoom)

The following pages have text that overflows the screen:

- [x] Get Started > Testing with Jest and React Testing Library (the Code Preview component variant used here might be the issue, it doesn't wrap nicely like the Components pages one does)
- [x] Get Started > Testing with Cypress (same as above?)
- [x] Get Started > Migrating to V3 (might be the `<code>` elements that won't wrap?)
- [x] Accessibility > Auditory Needs, Motor Needs, Visual Needs (the table with examples of impairments only just doesn't fit, looks like it's padding/margin related)
- [x] Accessibility > Invisible Illness, Neurodiversity (Long headings that won't wrap)
- [x] Accessibility > Subtitling, Transcripts and Audio Description (same issue as above?)
- [x] Styles > Colour (looks like the ic-data-list is to blame. Suggestion: if the list can't wrap, consider it essential - like how tables are allowed to scroll horizontally - and focus on making the page text wrap)
- [ ] Styles > Icon Groups (the bespoke component on this page doesn't play well at all with the narrow viewport. Suggestion: replace it with Tabs. They wrap great.)
- [x] Styles > Layout and Spacing (tables only just not fitting. Could be fixed akin to the Accessibility > Needs pages)
- [x] Components > Types (ic-data-list not quite fitting on the page)
- [x] Patterns > Top/Side Navigation Layout (the Code Preview component variant used here might be the issue, it only just doesn't wrap nicely like the Components pages one does)
- [x] Patterns > App Errors (the Component Previews seem to be a tiny bit too wide)
- [ ] Patterns > Form Validation, Success Messages (same as above, but worse. Components inside the previews spill outward) 
- [x] Community > Glossary of Terms (same as Styles > Colour, consider the table essential and focus on making the rest of the page text wrap)

 
### 🪜 How to reproduce

Tell us the steps to reproduce the problem:

1. Set browser viewport to 320px wide. (recommend opening dev tools in side-view and expanding it till page is desired width)
2. Visit affected pages
4. See the errors described in brackets

### 📸 Screenshots or code

Example of page failing: 

<img width="399" height="946" alt="Image" src="https://github.com/user-attachments/assets/8cef201c-9aeb-4665-a551-8f428adbc1c2" />

### 🚨 Urgency (low, medium or high)

High - this is an accessibility failure, and we'll have an exception on our Conformance Report until these are resolved.
